src/Providers/Maps/Google/Features/Shape.ts

```diff
@@ -0,0 +1,147 @@
+                this._markerCoordinates = new google.maps.LatLng(
+                    this._markerCoordinates.Lat,
+                    this._markerCoordinates.Lng
+                );
+
+                // Check if marker is inside shape based on shape element
+                if (shapeId) {
+                    this._shape =
+                        OutSystems.Maps.MapAPI.ShapeManager.GetShapeById(
+                            shapeId
+                        );
+
+                    // Check if shape contains marker based on shape type
+                    switch (this._shape.type) {
+                        case OSFramework.Maps.Enum.ShapeType.Polyline:
+                            // The Polyline is an unsupported shape to use the ContainsLocation API
+                            this._returnObjCode =
+                                OSFramework.Maps.Enum.Unsupported.code;
+                            this._returnObjMessage =
+                                OSFramework.Maps.Enum.Unsupported.message;
+                            break;
+                        case OSFramework.Maps.Enum.ShapeType.Rectangle:
+                            this._shapeRectangle();
+                            break;
+                        case OSFramework.Maps.Enum.ShapeType.Circle:
+                            this._shapeCircle();
+                            break;
+
+                        default: {
+                            this._shapeDefault();
+                        }
+                    }
+                } else {
+                    const shapeCoordinatesList = JSON.parse(
+                        coordinatesList.toLowerCase()
+                    );
+
+                    // To create a shape we need at least 3 coordinates
+                    if (shapeCoordinatesList.length >= 3) {
+                        const newShape = new google.maps.Polygon({
+                            paths: shapeCoordinatesList
+                        });
+
+                        // Set hidden a shape into the existing map to validate if the marker is inside
+                        newShape.setMap(null);
+                        this._isInsideShape =
+                            google.maps.geometry.poly.containsLocation(
+                                this._markerCoordinates,
+                                newShape
+                            );
+                        this._returnObjSuccess = true;
+                        this._returnObjCode =
+                            OSFramework.Maps.Enum.Success.code;
+                        this._returnObjMessage = this._isInsideShape.toString();
+                    }
+                }
+            }
+
+            // Set the return message to expose values
+            return {
+                isSuccess: this._returnObjSuccess,
+                code: this._returnObjCode,
+                message: this._returnObjMessage
+            };
+        }
+
+        public dispose(): void {
```

Comment:
We can delete this after delete the implementation.

Reply:
Done

Reply:
done